### ISSUE TITLE
After installation according to https://github.com/carrier-io/carrier-io/wiki/Quick-Start, jmeter demo fails due to get_influx_host function issue

### ISSUE THREAD

**Issue by dzavalei** [carrier-io/carrier-io#7]
After going through the installation steps, I am running into an error in Jenkins when triggering Jmeter standalone demo job (see stacktrace below).

It looks like following function creates a problem:

def get_influx_host(String env_var) {
    def match = env_var =~ 'http://(.+)/jenkins'
    return match[0][1]
}

I mean to say regex is not found, maybe env_var variable doesn't contain http://, but it would really help debug problem further if exception was caught and env_var was printed out with meaningful comment.

Started by user unknown or anonymous
Obtained demo_jenkins/Jenkinsfile_perui_demo from git https://github.com/carrier-io/carrier-io.git
Running in Durability level: MAX_SURVIVABILITY
[Pipeline] Start of Pipeline
[Pipeline] node
Running on Jenkins in /var/jenkins_home/workspace/demo_perfui
[Pipeline] {
[Pipeline] stage
[Pipeline] { (configure)
[Pipeline] deleteDir
[Pipeline] sh
+ mkdir reports
[Pipeline] sh
+ mkdir tests
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
java.lang.IndexOutOfBoundsException: index is out of range 0..-1 (index = 0)
	at org.codehaus.groovy.runtime.StringGroovyMethods.getAt(StringGroovyMethods.java:1384)
	at org.codehaus.groovy.runtime.dgm$1081.doMethodInvoke(Unknown Source)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
	at org.codehaus.groovy.runtime.callsite.PojoMetaClassSite.call(PojoMetaClassSite.java:47)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.kohsuke.groovy.sandbox.impl.Checker$10.call(Checker.java:425)
	at org.kohsuke.groovy.sandbox.GroovyInterceptor.onGetArray(GroovyInterceptor.java:124)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onGetArray(SandboxInterceptor.java:473)
	at org.kohsuke.groovy.sandbox.impl.Checker$10.call(Checker.java:422)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetArray(Checker.java:427)
	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.getArray(SandboxInvoker.java:45)
	at com.cloudbees.groovy.cps.impl.ArrayAccessBlock.rawGet(ArrayAccessBlock.java:21)
	at WorkflowScript.get_influx_host(WorkflowScript:3)
	at WorkflowScript.run(WorkflowScript:12)
	at ___cps.transform___(Native Method)
	at com.cloudbees.groovy.cps.impl.PropertyishBlock$ContinuationImpl.get(PropertyishBlock.java:74)
	at com.cloudbees.groovy.cps.LValueBlock$GetAdapter.receive(LValueBlock.java:30)
	at com.cloudbees.groovy.cps.impl.PropertyishBlock$ContinuationImpl.fixName(PropertyishBlock.java:66)
	at sun.reflect.GeneratedMethodAccessor91.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
	at com.cloudbees.groovy.cps.impl.ConstantBlock.eval(ConstantBlock.java:21)
	at com.cloudbees.groovy.cps.Next.step(Next.java:83)
	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
	at org.codehaus.groovy.runtime.GroovyCategorySupport$ThreadCategoryInfo.use(GroovyCategorySupport.java:129)
	at org.codehaus.groovy.runtime.GroovyCategorySupport.use(GroovyCategorySupport.java:268)
	at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:163)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$001(SandboxContinuable.java:18)
	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:51)
	at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:186)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:370)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$200(CpsThreadGroup.java:93)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:282)
	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:270)
	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:66)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:131)
	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Finished: FAILURE

**Comment by arozumenko** [comment]
@dzavalei have you fixed an errors after jenkins installation?

You suppose to have ~4 red errors in jenkins, one of them due to miss-configured JENKINS_URL
You need to open settings and save it. Jenkins URL is there, but not applied - Jenkins glitch :)

**Comment by dzavalei** [comment]
Thanks, @arozumenko !
It helped, however when using this web app listening on 9999 port for installation I still didn't get grafana dashboards created (I tried Amazon Linux 2 and Ubuntu 18.04 on AWS - no luck at all).
Only when I used https://github.com/carrier-io/carrier-io/blob/master/ubuntu/install.sh minimal functionality (Jenkins, Jmeter, InfluxDB, grafana) worked out of the box.
And still even after doing all this I had to follow your advice.
